dpgen2/entrypoint/submit.py

```diff
@@ -370,6 +370,14 @@ def workflow_concurrent_learning(
             if old_style
             else config["train"].get("init_models_paths", None)
         )
+        numb_models = (
+            config["numb_models"] if old_style else config["train"]["numb_models"]
+        )
+        if init_models_paths is not None and len(init_models_paths) != numb_models:
+            raise RuntimeError(
+                f"{len(init_models_paths)} init models provided, which does "
+                "not match numb_models={numb_models}"
+            )
```

Comment:
@Angel-Jia Please see the check here. It does not make sense if the length of the list does not matches number of models. therefore, we raise error when an empty list is provided.